code/modules/mob/new_player/poll.dm

```diff
@@ -243,10 +360,128 @@
 		. = client.holder.rank
 	return .
 
+
+/mob/new_player/proc/vote_rig_check()
+	if (usr != src)
+		if (!usr || !src)
+			return 0
+		//we gots ourselfs a dirty cheater on our hands!
+		log_game("[usr]([usr.ckey]) attempted to rig the vote by voting as [ckey]")
+		message_admins("[usr]([usr.ckey]) attempted to rig the vote by voting as [ckey]")
+		usr << "You don't seem to be [ckey]."
+		src << "Something went horribly wrong processing your vote. Please contact an administrator, they should have gotten a message about this"
+		return 0
+	return 1
+
+/mob/new_player/proc/vote_valid_check(pollid, holder, type)
+	pollid = text2num(pollid)
+	if (!pollid || pollid < 0)
+		return 0
+	//validate the poll is actually the right type of poll and its still active
+	var/DBQuery/select_query = dbcon.NewQuery({"
+		SELECT id
+		FROM [format_table_name("poll_question")]
+		WHERE
+			[(holder ? "" : "adminonly = false AND")]
+			id = [pollid]
+			AND
+			Now() BETWEEN starttime AND endtime
+			AND
+			polltype = '[type]'
+	"})
+	if (!select_query.Execute())
+		var/err = select_query.ErrorMsg()
+		log_game("SQL ERROR validating poll via poll_question table. Error : \[[err]\]\n")
+		return 0
+	if (!select_query.NextRow())
+		return 0
+	return 1
+
+/mob/new_player/proc/vote_on_irv_poll(pollid, list/votelist)
+	if (!vote_rig_check())
+		return 0
+	pollid = text2num(pollid)
+	if (!pollid || pollid < 0)
+		return 0
+	if (!votelist || !istype(votelist) || !votelist.len)
+		return 0
+	if (!client)
+		return 0
+	if(!dbcon.IsConnected())

```

Comment:
I don't think this is in any of the other `vote_on_x` procs, could you add it?
Also would it be better to have this as the first code ran so everything above it can be skipped if there's no DB?